### ISSUE TITLE
Help adding new metrics for output in regression tasks

### ISSUE THREAD

**Issue by danielalthoff** [topepo/caret#837]
Hello, 
I've read issue #657 and wondered if there is a way to add new metrics.
I'm not sure adding random metrics is desired for everyone for the caret package, but I wondered if it is possible to add metrics functions on my own.

Example: is it possible to override postResample ? : 
"    me <- mean(pred - obs)
      mse <- mean((pred - obs)^2)
      mae <- mean(abs(pred - obs))
      out <- c(sqrt(mse), resamplCor^2, mae, me)
    }
    names(out) <- c("RMSE", "Rsquared", "MAE", "ME")
  }"

The code above was successful while using train() function, but when I use a function from another package that imports caret, postResample, it uses the OLD postResample function.

I guess it has the function already imported. Is there a way to re-import that function? make the other package import the new postResample?

Thank you.

**Comment by topepo** [comment]
See [Section 5.5.5](https://topepo.github.io/caret/model-training-and-tuning.html#customizing-the-tuning-process)

**Comment by danielalthoff** [comment]
The real problem I am having is:

I changed the postResample and when I use train, I get the new performance metrics I need.. but when I use another function to call caret... it works as the old function.

**Comment by topepo** [comment]
That's not really the real issue. `postResample` is an internal `caret` function and by making another copy in your local environment, it is easy for the `caret` version to mask your local copy. Basically, it is an approach that is unlikely to work in general.

**Comment by danielalthoff** [comment]
Is there an alternative that might work?
I checked Section 5.5.5... " can also compute custom performance metrics. The trainControl function has a argument called summaryFunction that specifies a function for computing performance. "  In tranControl.R:
                         summaryFunction = defaultSummary,

But summaryFunction is only called in aaa.R.. and I wasn't able to find much from there:
defaultSummary <- function(data, lev = NULL, model = NULL)
{
  if(is.character(data$obs)) data$obs <- factor(data$obs, levels = lev)
  postResample(data[,"pred"], data[,"obs"])
}

that's why I tried altering postResample. Am I missing something from Section 5.5.5? Is there a way I can alter the function (not only in my local envirionment) and get different performance metrics/index?
Because the function that requires caret keeps using the old one.
Thank you.
